src/text_manipulation.py

```diff
@@ -0,0 +1,112 @@
+import random
+import spacy
+import utils
+
+class Manipulator:
+
+    def __init__(self):
+        self.nlp = spacy.load('en_core_web_sm')
+
+    def reorder_paragraphs(self, paragraph_list, random_seed=3142):
+
+        random.seed(random_seed)
+
+        paragraph_nested_list = []
+
+        for this_paragraph in paragraph_list:
+            paragraph_nested_list.append([this_paragraph])
+
+        random.shuffle(paragraph_nested_list)
+
+        return paragraph_nested_list
+
+    def print_nested_list(self, input_list):
+
+        for this_item in input_list:
+            print(this_item[0])
+            print(" ")
+
+    def print_list(self, input_list):
+
+        for this_item in input_list:
+            print(this_item)
+            print(" ")
+
+    #def swap_two_sentences(self, sentences_list):
+
+     #   idx = range(len(sentences_list))
+     #  i1, i2 = random.sample(idx, 2)
+     # sentences_list[i1], sentences_list[i2] = sentences_list[i2], sentences_list[i1]
+
+     #return sentences_list
+
+    def reorder_sentences(self, sentences_list):
+        # the below function takes a random sentence in a paragraph and replaces it a random different point
+
+        # the line below finds the length of the paragraph
+        length_of_paragraph = len(sentences_list)
+
+        # the line below creates a copy of the original list of text
+        copy_sentences_list = sentences_list.copy()
```

Comment:
I believe the reason you are making a copy is that you want to have some memory of the correct answer. This is a good idea, but since the answer is basically the original text I think we should store it as a string here before we do anything (remember strings are immutable so you won't be able to accidentally change it, which is what we want). Still a good idea to make a copy sine python appears to pass variables as references to functions so editing a mutable object will actually edit it outside the function to which you never want.

Reply:
So you need to work out how to join (<- hint) a list of strings into one string

Reply:
then also copy the list, and never touch the original just edit the copy